src/main/java/de/azapps/kafkabackup/common/record/RecordJSONSerde.java

```diff
@@ -54,14 +57,27 @@ public Record deserialize(JsonParser jp, DeserializationContext ctx) throws IOEx
             String topic = node.get(TOPIC_PROPERTY).asText();
             int partition = node.get(PARTITION_PROPERTY).asInt();
             long offset = node.get(OFFSET_PROPERTY).asLong();
+            String timestampTypeStr = node.get(TIMESTAMP_TYPE_PROPERTY).asText(null); // Default seems to be the string "null", which is not wanted here
+            TimestampType timestampType;
+            if (timestampTypeStr != null) {
+                timestampType = TimestampType.forName(timestampTypeStr);
+            } else {
+                timestampType = TimestampType.NO_TIMESTAMP_TYPE;
+            }
+            Long timestamp; // used instead of primitive long to use it as an Optional with null as a legal value
+            if (node.hasNonNull(TIMESTAMP_PROPERTY)) {
+                timestamp = node.get(TIMESTAMP_PROPERTY).asLong();
+            } else {
+                timestamp = null;
+            }
```

Comment:
same here - let's maybe try to refactor using Optional and .map ? 🤔

Reply:
Had to use slightly different solution (since Record constructor takes Long, not Optional), but still cleaner